api/src/test/java/demo/service/validation/filter_validator/FilterValidatorDateTest.java

```diff
@@ -0,0 +1,87 @@
+package demo.service.validation.filter_validator;
+
+import demo.controller.dto.FilterDto;
+import demo.service.filter.DataTypes;
+import demo.service.filter.DateCriteria;
+import demo.service.filter.NumberCriteria;
+import demo.service.validation.ValidationResult;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.MethodSource;
+import org.mockito.MockitoAnnotations;
+
+import java.time.Instant;
+import java.util.stream.Stream;
+
+import static demo.service.validation.filter_validator.FilterErrors.INVALID_CRITERIA_ERROR;
+import static demo.service.validation.filter_validator.FilterErrors.INVALID_DATE_VALUE;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertNull;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+class FilterValidatorDateTest {
+
+    private FilterDtoValidator filterDtoValidator;
+
+    @BeforeEach
+    void setUp() {
```

Comment:
The given code 
`
@ExtendWith(MockitoExtension.class)
class FilterValidatorDateTest {

    private final FilterDtoValidator filterDtoValidator;
`

gave a error  'filterDtoValidator' might not have been initialized, and intellj wants to add a constructor

Reply:
since the field is final, it has to be initialized in the constructor or in the same line with the declaration, so you can do 
`private final FilterDtoValidator filterDtoValidator = new FilterDtoValidator();`
which looks a lot more like the code that we usually write. Currently we have very specific test initialization happening. 

EDIT: as you are not using mocking here, it should be removed alltogether!